azurerm_eventhub/main.tf

```diff
@@ -44,20 +44,21 @@ resource "azurerm_eventhub_namespace" "eventhub_ns" {
 }
 
 resource "azurerm_eventhub" "eventhub" {
-  name                = local.resource_name
+  count               = length(var.eventhubs)
+  name                = var.eventhubs[count.index].name
   namespace_name      = azurerm_eventhub_namespace.eventhub_ns.name
   resource_group_name = var.resource_group_name
-  partition_count     = var.partition_count
-  message_retention   = var.message_retention
+  partition_count     = var.eventhubs[count.index].partition_count
+  message_retention   = var.eventhubs[count.index].message_retention
 }
 
 resource "azurerm_eventhub_authorization_rule" "eventhub_rule" {
   count               = length(var.eventhub_authorization_rules)
```

Comment:
Use for_each instead of count to avoid problems when adding new items to the list

Reply:
If I am going to apply this change I will be forced to destroy the current *hubs which are already in use:
```
  # azurerm_eventhub.eventhub["io-p-evh-siem"] will be created
  + resource "azurerm_eventhub" "eventhub" {
      + id                  = (known after apply)
      + message_retention   = 5
      + name                = "io-p-evh-siem"
      + namespace_name      = "io-p-evhns-siem"
      + partition_count     = 4
      + partition_ids       = (known after apply)
      + resource_group_name = "io-p-rg-siem"
    }

  # azurerm_eventhub.eventhub["io-p-evh-siem-logs"] will be created
  + resource "azurerm_eventhub" "eventhub" {
      + id                  = (known after apply)
      + message_retention   = 5
      + name                = "io-p-evh-siem-logs"
      + namespace_name      = "io-p-evhns-siem"
      + partition_count     = 4
      + partition_ids       = (known after apply)
      + resource_group_name = "io-p-rg-siem"
    }

Plan: 2 to add, 0 to change, 2 to destroy.
```